client_test/helpers_test.go

```diff
@@ -23,41 +23,47 @@ const defaultTimeout = time.Second
 // waitWithTimeoutForCompletedObjectiveIds waits up to the given timeout for completed objectives and returns when the all objective ids provided have been completed.
 // If the timeout lapses and the objectives have not all completed, the parent test will be failed.
 func waitTimeForCompletedObjectiveIds(t *testing.T, client *client.Client, timeout time.Duration, ids ...protocols.ObjectiveId) {
-	waitAndSendOn := func(allDone chan interface{}) {
-		waitForCompletedObjectiveIds(client, ids...)
-		allDone <- struct{}{}
-	}
-	allDone := make(chan interface{})
-	go waitAndSendOn(allDone)
 
-	select {
-	case <-time.After(timeout):
-		t.Fatalf("Objective ids %s failed to complete in one second on client %s", ids, client.Address)
-	case <-allDone:
-		return
+	waitAndSendOn := func(completed map[protocols.ObjectiveId]bool, allDone chan interface{}) {
+
+		// We continue to consume completed objective ids from the chan until all have been completed
+		for got := range client.CompletedObjectives() {
+			// Mark the objective as completed
+			completed[got] = true
+
+			// If all objectives are completed we can send the all done signal and return
+			isDone := true
+			for _, objectiveCompleted := range completed {
+				isDone = isDone && objectiveCompleted
+			}
+			if isDone {
+				allDone <- struct{}{}
+				return
+
+			}
+		}
+
 	}
-}
 
-// waitForCompletedObjectiveIds waits for completed objectives and returns when the all objective ids provided have been completed.
-func waitForCompletedObjectiveIds(client *client.Client, ids ...protocols.ObjectiveId) {
-	// Create a map of all objective ids to wait for and set to false
+	allDone := make(chan interface{})
+	// Create a map to keep track of completed objectives
 	completed := make(map[protocols.ObjectiveId]bool)
 	for _, id := range ids {
 		completed[id] = false
 	}
```

Comment:
Outside scope of this PR but I believe these lines are unnecessary -- the map will return a default (zero) value of `false` for unintialized keys.

Reply:
Good point, updated in [ed05bab](https://github.com/statechannels/go-nitro/pull/379/commits/ed05babd81a3080773c6665961556b2b5ccddb76)